weave/workers/weave_scorer.py

```diff
@@ -0,0 +1,445 @@
+
+    score_call = _get_score_call(score_call_id, project_id)
+
+    call_ref = InternalCallRef(project_id=project_id, id=call.id)  # type: ignore
+    score_call_ref = InternalCallRef(project_id=project_id, id=score_call.id)  # type: ignore
+
+    results_json = to_json(result, project_id, None)  # type: ignore
+    payload = {"output": results_json}
+
+    server = get_trace_server()
+
+    feedback_req = FeedbackCreateReq(
+        project_id=project_id,
+        weave_ref=call_ref.uri(),
+        feedback_type=RUNNABLE_FEEDBACK_TYPE_PREFIX + "." + scorer.name,  # type: ignore
+        payload=payload,
+        runnable_ref=scorer.__dict__["internal_ref"].uri(),
+        call_ref=score_call_ref.uri(),
+        wb_user_id=wb_user_id,
+        trigger_ref=monitor_internal_ref.uri(),
+    )
+
+    server.feedback_create(feedback_req)
+
+
+async def process_ended_call(ended_call: tsi.EndedCallSchemaForInsert) -> None:
+    project_id = ended_call.project_id
+
+    active_monitors_ref_user_ids = get_active_monitors(project_id)
+
+    for monitor, monitor_internal_ref, wb_user_id in active_monitors_ref_user_ids:
+        await process_monitor(monitor, monitor_internal_ref, ended_call, wb_user_id)  # type: ignore
+
+
+def _call_processor_done_callback(
+    task: asyncio.Task,
+    msg: Message,
+    consumer: KafkaConsumer,
+    call_processors: set[asyncio.Task],
+) -> None:
+    call_processors.discard(task)
+
+    try:
+        task.result()
+        consumer.commit(msg)
+    except Exception as e:
+        logger.exception(
+            f"Error processing message: {e.__class__.__name__} {e}", exc_info=e
+        )
+
+
+async def process_kafka_message(
+    msg: Message, consumer: KafkaConsumer, call_processors: set[asyncio.Task]
+) -> bool:
+    """Process a single Kafka message and create a task for it."""
+    try:
+        ended_call = tsi.EndedCallSchemaForInsert.model_validate_json(
+            msg.value().decode("utf-8")
+        )
+        logger.info("Processing ended call %s", ended_call.id)
+
+        task = asyncio.create_task(process_ended_call(ended_call))
+        call_processors.add(task)
+        task.add_done_callback(
+            lambda t: _call_processor_done_callback(t, msg, consumer, call_processors)
+        )
+
+        return True  # noqa: TRY300
+    except Exception as e:
+        logger.exception("Error processing message: %s", e, exc_info=e)
+        return False
+
+
+async def handle_kafka_errors(msg: Message) -> bool:
+    """Handle Kafka-specific errors."""
+    if msg.error():
+        if msg.error().code() == KafkaError._PARTITION_EOF:
+            logger.error(
+                "%% %s [%d] reached end at offset %d\n"
+                % (msg.topic(), msg.partition(), msg.offset())
+            )
+        else:
+            logger.exception("Kafka error: %s", msg.error())
+
+        return False
+
+    return True
+
+
+async def cleanup_tasks(call_processors: set[asyncio.Task]) -> set[asyncio.Task]:
```

Comment:
this function name seems misleading. reading the code, it is more like: `waitAndPopFirstComleted`